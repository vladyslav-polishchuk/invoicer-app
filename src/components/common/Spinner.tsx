import { Backdrop, CircularProgress, Typography } from '@mui/material';

export default function Spinner() {
  return (
    <Backdrop
      sx={{
        backgroundColor: 'rgba(255, 255, 255, 0.6)',
        zIndex: 999,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
      }}
      open
    >
      <Typography variant="h5" sx={{ mb: 2 }}>
        Please wait while we prepare things for you
      </Typography>
      <CircularProgress size={60} />
    </Backdrop>
  );
}
