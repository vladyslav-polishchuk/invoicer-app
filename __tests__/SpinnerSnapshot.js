import { render } from '@testing-library/react';
import Spinner from '../src/components/common/Spinner';

it('Should render spinner', () => {
  const { container } = render(<Spinner />);

  expect(container).toMatchSnapshot();
});
