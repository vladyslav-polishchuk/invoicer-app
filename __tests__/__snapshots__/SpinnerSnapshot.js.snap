// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render spinner 1`] = `
<div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-5ou4mu-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ch2k2f-MuiTypography-root"
    >
      Please wait while we prepare things for you
    </h5>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
